### ISSUE TITLE
Controllable Text-to-Image Generation with GPT-4

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#219]
# TL; DR

- Generate Tikz sketch with GPT-4 for compositionality, and I2I it with Control Net Diffusion..

# Motivation

- Diffusion models tend to lack compositionality and the capacity for spatial relationship parsing (e.g., left, right, above, below), while LLMs excel at these concepts. 

# Formulation

<img width="938" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/4784f391-1127-4d48-9b04-c7a13d320863">

- Combine LLM with Control-Net; generate sketch layout with LLM (code) and use it as semantic guidance for diffusion.
- As there is no large scale (sketch, image) dataset publically available, generate poligonal layout from real images of COCO or LVIS.

<img width="484" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/ecc71781-e931-44a2-86ad-1e48b9851565">

- Additionally prompt GPT-4 to output grounding tokens (object name and its location) so that the diffusion model does not miscorrelate sketches of multiple entities.

<img width="934" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/cf8b0782-28c8-4200-ba37-c498832b7b33">

# Evaluation

<img width="902" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/5203ea8e-bb2e-4985-9b0b-dea30ff23fc0">
<img width="935" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/7059e4fe-1e68-481c-b676-42ccee2cdfe9">
<img width="936" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/496850e8-a36b-4f5c-bd13-7d3445fd3a80">
<img width="951" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/65e3034f-18ab-4e5f-aeb2-bf108c1dd7b8">
<img width="930" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/d16c284d-2d9d-44a2-af86-9119226502ff">
<img width="933" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/0af4e1a3-0299-4be9-9656-6eacda41f9dd">

# Inspiration

- LLMs and Diffusion models, two pillars of modern **Gen AI**, can complement each other for visual creation tasks.
